res/controllers/Reloop-Beatmix-2-4-scripts.js

```diff
@@ -0,0 +1,773 @@
+                            true);
+                    } else {
+                        engine.setValue("[Playlist]", "SelectNextPlaylist",
+                            true);
+                    }
+                }
+            }
+            break;
+        case 2: // Track mode
+            engine.setValue("[Playlist]", "SelectTrackKnob", value * step);
+            break;
+        case 3: // Preview mode
+            engine.setValue("[PreviewDeck1]", "playposition", Math.max(0,
+                Math.min(1, engine.getValue("[PreviewDeck1]",
+                    "playposition") + 0.02 * value * step)));
+            break;
+    }
+};
+
+ReloopBeatmix24.TraxTurn = function(channel, control, value, status, group) {
+    ReloopBeatmix24.traxSelect(value - 0x40, 1);
+};
+
+ReloopBeatmix24.ShiftTraxTurn = function(channel, control, value, status, group) {
+    ReloopBeatmix24.traxSelect(value - 0x40, 10);
+};
+
+ReloopBeatmix24.TraxPush = function(channel, control, value, status, group) {
+    switch (traxMode) {
+        case 1: // Playlist mode
+            engine.setValue("[Playlist]", "ToggleSelectedSidebarItem",
+                value);
+            break;
+        case 2: // Track mode
+            engine.setValue("[PreviewDeck1]", "LoadSelectedTrackAndPlay",
+                value ? 1 : 0);
+            traxMode = 3;
+            break;
+        case 3: // Preview mode
+            if (value == DOWN) {
+                engine.setValue("[PreviewDeck1]", "play", engine.getValue(
+                    "[PreviewDeck1]", "play") ? 0 : 1);
+            }
+            break;
+    }
+};
+
+ReloopBeatmix24.ShiftTraxPush = function(channel, control, value, status, group) {
+    ReloopBeatmix24.TraxPush(channel, control, value, status, group);
+};
+
+ReloopBeatmix24.BackButton = function(channel, control, value, status, group) {
+    if (value == DOWN) {
+        switch (traxMode) {
+            case 1: // Playlist mode
+                traxMode = 2; // Switch to track mode
+                break;
+            case 2: // Track mode
+                traxMode = 1; // Switch to playlist mode
+                break;
+            case 3: // Preview mode
+                traxMode = 2; // Switch to track mode
+                break;
+        }
+    }
+};
+
+ReloopBeatmix24.SamplerLoadEject = function(channel, control, value, status,
+    group) {
+    if (value == DOWN) {
+        if (engine.getValue(group, "track_samples")) { // Loaded
+            engine.setValue(group, "eject", 1);
+        } else { // Empty
+            engine.setValue(group, "LoadSelectedTrack", 1);
+        }
+    } else { // UP
+        if (engine.getValue(group, "track_samples")) { // Loaded
+            engine.setValue(group, "LoadSelectedTrack", 0);
+        } else { // Empty
+            engine.setValue(group, "eject", 0);
+        }
+    }
+};
+
+ReloopBeatmix24.SamplerVol = function(channel, control, value, status, group) {
+    for (var i = 1; i <= 4; i++) {
```

Comment:
Instead of a fixed 1-4, use [Master], num_samplers here in case more samplers are in use. AFAIK that control won't change unless a script sets it (or maybe if a skin with more samplers is loaded?), so for efficiency I think it'd be a good idea to save the result of engine.getValue("[Master]", "num_samplers") in a variable and reference the variable here.